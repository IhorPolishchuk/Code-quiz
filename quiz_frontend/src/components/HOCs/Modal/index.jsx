import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Button } from './../../ui';
import icons from './../../../utils/icons.json';

import styles from './index.sass';

class Modal extends Component {
    render () {
        const { title, children } = this.props;
        return (
            <div
                className={styles.modal__wrapper}
            >
                <div
                    className={styles.modal__body}
                >
                    <div
                        className={styles.modal_top__wrapper}
                    >
                        <div
                            className={styles.modal__title}
                        >
                            {title}
                        </div>
                        <Button
                            icon={icons['cancel-circle']}
                            className={`${styles.modal__close} ${styles.modal__button}`}
                        />
                    </div>
                    <div
                        className={styles.modal__content}
                    >
                        {children}
                    </div>
                    <hr/>
                    <div
                        className={styles.modal_bottom__wrapper}
                    >
                        <Button
                            title={'Cancel'}
                            icon={icons['cancel-circle']}
                            className={styles.modal__button}
                        />
                        <Button
                            title={'Ð¡onfirm'}
                            icon={icons.checkmark}
                            className={styles.modal__button}
                        />
                    </div>
                </div>
            </div>
        );
    }
}

Modal.PropTypes = {
    title: PropTypes.string.isRequired,
    children: PropTypes.any
};

export default Modal;
