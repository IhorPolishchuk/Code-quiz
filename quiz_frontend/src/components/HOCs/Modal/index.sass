$modal_border_color: #c9c9c9
$modal_body_bg: #ffffff
$modal_wrapper_color: rgba(0,0,0,0.7)
.modal
    &__wrapper
        position: fixed
        left: 0
        top: 0
        width: 100%
        height: 100vh
        background-color: $modal_wrapper_color
    &__body
        margin: 0 auto
        margin-top: 100px
        max-width: 900px
        min-width: 400px
        padding: 10px 10px
        border-radius: 15px
        border: 1px solid $modal_border_color
        background-color: $modal_body_bg
    &__content
        max-height: 400px
        overflow-y: scroll
        padding: 10px 10px
    &_bottom__wrapper, &_top__wrapper
        display: flex
        flex-direction: row
        justify-content: space-between
        align-items: center
        padding: 10px 10px
    &__button
        font-family: RobotoMedium
    &__close
        padding: 5px 10px
    &__title
        font-family: RobotoBold
        font-size: 26px