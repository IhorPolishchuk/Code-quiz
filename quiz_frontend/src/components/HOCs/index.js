export { default as ScrollToTopComponent } from './ScrollToTopComponent';
export { default as PrivateRoute } from './PrivateRoute';
