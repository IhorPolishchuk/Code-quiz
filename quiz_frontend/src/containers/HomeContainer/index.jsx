import React, { Component } from 'react';

import { TextField } from'./../../components/ui';

class HomeContainer extends Component {
    render () {
        return (
            <div>
                <TextField
                    label="text"
                    error={false}
                    regex={/\.com/g}
                    errorMessage="Should contain '.com'"
                />
            </div>
        );
    }
}

export default HomeContainer;
