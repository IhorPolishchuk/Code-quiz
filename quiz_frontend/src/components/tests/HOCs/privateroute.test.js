import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HomeContainer from './../../../containers/HomeContainer';
import { PrivateRoute } from './../../HOCs';
/**
 * ToDo: do test routing
 */
describe('PrivateRoute HOC test', () => {
    it('PrivateRoute will redirect to /signin route', () => {
        const pr = shallow(
            <PrivateRoute
                to='/'
                authed={false}
                component={HomeContainer}
            />
        );
        const tree = toJson(pr);

        expect(tree).toMatchSnapshot();
    });
    it('PrivateRoute will pass and render component', () => {
        const pr = shallow(
            <PrivateRoute
                to='/'
                authed
                component={HomeContainer}
            />
        );
        const tree = toJson(pr);

        expect(tree).toMatchSnapshot();
    });
});
