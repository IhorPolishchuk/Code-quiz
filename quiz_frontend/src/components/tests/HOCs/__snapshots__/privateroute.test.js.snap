// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute HOC test PrivateRoute will pass and render component 1`] = `
<Route
  render={[Function]}
  to="/"
/>
`;

exports[`PrivateRoute HOC test PrivateRoute will redirect to /signin route 1`] = `
<Route
  render={[Function]}
  to="/"
/>
`;
